mix.exs

```diff
@@ -12,6 +12,8 @@ defmodule Floki.Mixfile do
       description: @description,
       elixir: "~> 1.8",
       package: package(),
+      erlc_paths: ["src", "gen"],
+      compilers: [:gleam | Mix.compilers()],
```

Comment:
This PR now works in my Gleam project, however these changes break it in an Elixir project. perhaps not surprisingly.

```
** (Mix) The task "compile.gleam" could not be found. Did you mean "compile.erlang"?
```

I think the solution is to only add the gleam compiler option if it is required by the calling library.

The options I have thought of are.
- An env variable 
- add the compiler if the module `Mix.Gleam` is defined, because using https://github.com/gleam-lang/mix_gleam is how gleam is added to mix projects. 
- some other option in mix dependencies I haven't though of.